src/main/java/gift/service/KakaoOAuthService.java

```diff
@@ -0,0 +1,62 @@
+package gift.service;
+
+import gift.dto.KakaoTokenResponse;
+import gift.dto.KakaoUserResponse;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.http.*;
+import org.springframework.stereotype.Service;
+import org.springframework.util.LinkedMultiValueMap;
+import org.springframework.util.MultiValueMap;
+import org.springframework.web.client.RestTemplate;
+
+@Service
+public class KakaoOAuthService {
+
+    @Value("${kakao.client-id}")
+    private String clientId;
+
+    @Value("${kakao.redirect-uri}")
+    private String redirectUri;
+
+    private final RestTemplate restTemplate = new RestTemplate();
+
+    public KakaoTokenResponse getToken(String code) {
+        MultiValueMap<String, String> params = new LinkedMultiValueMap<>();
+        params.add("grant_type", "authorization_code");
+        params.add("client_id", clientId);
+        params.add("redirect_uri", redirectUri);
+        params.add("code", code);
+
+        HttpHeaders headers = new HttpHeaders();
+        headers.setContentType(MediaType.APPLICATION_FORM_URLENCODED);
+
+        HttpEntity<MultiValueMap<String, String>> request = new HttpEntity<>(params, headers);
+
+        ResponseEntity<KakaoTokenResponse> response = restTemplate.exchange(
+                "https://kauth.kakao.com/oauth/token",
+                HttpMethod.POST,
+                request,
+                KakaoTokenResponse.class
+        );
```

Comment:
exchange 대신에 `getForEntity`, `postForEntity` 메서드를 대신 활용해보면 어떨까요? 😃

```suggestion
        ResponseEntity<KakaoTokenResponse> response = restTemplate.postForEntity("https://kauth.kakao.com/oauth/token", request, KakaoTokenResponse.class);
```